src/contracts/fungible/data/outcoins.rs

```diff
@@ -175,3 +177,23 @@ impl FromStr for ConsealCoins {
         }
     }
 }
+
+impl Eq for OutpointCoins {}
+
+impl PartialOrd for OutpointCoins {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for OutpointCoins {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.outpoint.cmp(&other.outpoint)
+    }
+}
+
+impl Hash for OutpointCoins {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.outpoint.hash(state);
+    }
+}
```

Comment:
This is wrong, since two distinct `OutpointCoins` with distinct amounts but the same output will be considered equal and will result in the same hash, replacing each other in any map

Reply:
the compiler required me to add these traits in order to define [this](https://github.com/LNP-BP/rgb-sdk/pull/31/files#diff-e5fe85b6ecefe2218ccf44990e99b82ccc92348897776f37eec3f97377306707R363-R364). I don't know how to properly implement them though, I had encountered some problems when trying to compare the `coins` parameter, since `AccountingValue` is defined as `f64` and so it doesn't have the `cmp` method defined. if you can fix them I will be really happy to learn how to properly implement these traits

Reply:
Ok, I see, the invoice deserialization must be done differently in SDK, so we do not need this here. And there is no right way of implementing `Eq` for `f64`, and this is the reason it is not implemented in rust; see more in the introduction in https://doc.rust-lang.org/std/cmp/trait.PartialEq.html

Reply:
this is not for invoice deserialization, this is to define a `HashSet<OutpointCoins>` (in the `issue` method). if you know how to define that object in the sdk without adding these traits, please show me (on LNP-BP/rgb-sdk#31).

Reply:
Hm, I think I was wrong there in defining that type as a `HashSet`, not `Vec`. It should rather be `HashMap<OutPoint, u64>`, since we need to make sure we do not duplicate outpoints for the issue control. Will make that part of the next release

Reply:
Ok, as far as I see the issue method takes correct data 

https://github.com/LNP-BP/rgb-node/blob/master/src/contracts/fungible/processor.rs#L36

... so it is `i9n` module that is incorrect:
https://github.com/LNP-BP/rgb-node/blob/master/src/i9n/fungible.rs#L76

Fixed that with https://github.com/LNP-BP/rgb-node/commit/c3a15e92051f20009888e70dd77c88eb44768480